web/src/invite_stream_picker_pill.ts

```diff
@@ -45,15 +45,35 @@ export function add_default_stream_pills(pill_widget: stream_pill.StreamPillWidg
     }
 }
 
-export function create($stream_pill_container: JQuery): stream_pill.StreamPillWidget {
+export function add_specific_stream_pills(
+    pill_widget: stream_pill.StreamPillWidget,
+    stream_ids: number[],
+): void {
+    for (const stream_id of stream_ids) {
+        const sub = stream_data.get_sub_by_id(stream_id);
+        if (sub) {
+            stream_pill.append_stream(sub, pill_widget, false);
+        }
+    }
+}
+
+export function create(
+    $stream_pill_container: JQuery,
+    stream_ids?: number[],
+    is_edit_invite_modal = false,
+): stream_pill.StreamPillWidget {
     const pill_widget = input_pill.create({
         $container: $stream_pill_container,
         create_item_from_text: create_item_from_stream_name,
         get_text_from_item: stream_pill.get_stream_name_from_item,
         generate_pill_html: stream_pill.generate_pill_html,
         get_display_value_from_item: stream_pill.get_display_value_from_item,
     });
-    add_default_stream_pills(pill_widget);
+    if (stream_ids && is_edit_invite_modal) {
+        add_specific_stream_pills(pill_widget, stream_ids);
+    } else {
+        add_default_stream_pills(pill_widget);
```

Comment:
I think this can just call `add_specific_stream_pills` once with `stream_ids` being passed as default stream IDs when creating invite and the original stream IDs list when editing. This way `is_edit_invite_modal` will not be needed and we would not need two functions `add_specific_stream_pills` and `add_default_stream_pills`.